qcodes/instrument/delegate/delegate_instrument.py

```diff
@@ -264,6 +275,99 @@ def _create_and_add_parameter(
             **kwargs
         )
 
+    def _create_and_add_channels(
+        self,
+        station: Station,
+        channels: Union[Mapping[str, Sequence[str]], Mapping[str, str]],
+    ) -> None:
+        """Add channels to the instrument.
+
+        .. code-block:: yaml
+
+        field:
+            type: qcodes.instrument.delegate.DelegateInstrument
+            init:
+            channels:
+                type: mypackage.mydevice.mydevice_channel.MyDeviceChannel
+                top_barrier:
+                  channel: dac.ch01
+                  my_input_param: 0
+
+        Args:
+            station: QCoDeS station, i.e. representation of the entire physical
+                setup.
+            channels: Mapping of names/aliases to instrument channels and
+                optional input parameters if a channel wrapper class is used to
+                instantiate the channel. If no 'type' field is given, the
+                channel is added as is using ``self.add_submodule``.
+
+        """
+
+        channel_wrapper = None
+        if 'type' in channels.keys():
+            module_name = '.'.join(channels['type'].split('.')[:-1])
+            instr_class_name = channels['type'].split('.')[-1]
+            module = importlib.import_module(module_name)
+            channel_wrapper = getattr(module, instr_class_name)
+
+        for param_name, input_params in channels.items():
+            if param_name != 'type':
+                self._create_and_add_channel(
+                    param_name=param_name,
+                    station=station,
+                    input_params=input_params,
+                    channel_wrapper=channel_wrapper,
+                )
+
+    def _create_and_add_channel(
+        self,
+        param_name: str,
+        station: Station,
+        input_params: Union[str, Mapping[str, Any]],
+        channel_wrapper: Optional[InstrumentChannel],
+        **kwargs: Any
+    ) -> None:
+        """Adds a channel to the instrument.
+
+        Args:
+            param_name: Alias/name of the channel.
+            station: QCoDeS' station containing the instrument containing the
+                channel.
+            input_params: Either the path to the channel or keyworded arguments
+                with 'channel' key containing the path of the channel and any
+                other input arguments taken by channel_wrapper.
+            channel_wrapper: Optional class to construct the channel. If none
+                given, the channel is added as is using ``self.add_submodule``.
+
+        """
+
+        if isinstance(input_params, str):
+            try:
+                instrument_name, channel_name = input_params.split('.')
```

Comment:
Are we sure this is always of the form `inst.channel` it could be `inst.channel.subchannel`

Reply:
Good point. Not sure, it's probably best to use parse_instrument_path in that case. I'll make the change.